src/codeu/chat/client/core/View.java

```diff
@@ -138,6 +138,29 @@ public View(ConnectionSource source) {
 		return messages;
 	}
 
+	@Override
+	public Collection<String> getBots() {
+
+		final Collection<String> allBots = new ArrayList<>();
+
+		try (final Connection connection = source.connect()) {
+
+			Serializers.INTEGER.write(connection.out(), NetworkCode.GET_ALL_BOTS_REQUEST);
+
+			if (Serializers.INTEGER.read(connection.in()) == NetworkCode.GET_ALL_BOTS_RESPONSE) {
+				allBots.add(Serializers.STRING.read(connection.in()));
+			} else {
+				LOG.error("Response from server failed.");
+			}
+
+		} catch (Exception ex) {
+			System.out.println("ERROR: Exception during call on server. Check log for details.");
+			LOG.error(ex, "Exception during call on server.");
+		}
+		//returns null if there are no bots available 
```

Comment:
It might make sense to just return the empty `allBots` list? That way code that calls this function doesn't have to check for `null`?